docs/block-processing/us.md

```diff
@@ -0,0 +1,358 @@
+and updates should be handled carefully.
+
+## Verification
+
+### Block header verification
+
+First of all, we check that block version of a verifiable block is _adopted_ or _competing_.
+
+### Update proposal verification
+
+#### General checks
+
+* _No duplicated proposer check_: that no one stakeholder sent two update proposals within current epoch. 
+We prohibit it because of opportunity spam update proposals from one stakeholder.
+
+* _Update proposal size check_: proposal doesn't exceed maximal proposal size.
+  * Update proposal size is computed as number of bytes in serialized `UpdateProposal`.
+  * Update proposal size limit is part of `BlockVersionData`.
+  * If the size of a proposal is greater than the limit from the adopted `BlockVersionData`, the proposal is invalid.
+
+* _Attributes known check_: if `verifyAllIsKnown` (defined [here](overall.md#verifyallisknown-flag))
+is `True`, all update proposal attributes must be known.
+ 
+* _Uniqueness of proposal check_: that there is no active proposal with the same id.
+
+* _Software version check_: numeric software version of application is 1 more 
+than of last confirmed proposal for this application.
+
+* _Stake to include into block check_: sum of positive votes for this proposal is enough 
+to be included into block (at least `updateProposalThd`).
+We wouldn't like adversary stakeholders to have opportunity to spam blockchain by `UpdateProposal`.
+So everyone can send update proposal to network, but it can get into block only along with enough votes.
+
+#### Protocol version checks
+
+* _Protocol version following check_: 
+proposed block version must adhere to the rules described [above](#competing-block-version).
+
+* _BlockVersionModifier consistenty check_: `BlockVersionModifier` from proposal is consistent 
+with `BlockVersionData` for adopted protocol version or 
+with `BlockVersionModifier` from _competing_ proposals with the same protocol version.  
+There are three cases to check:
+  * Block version from proposal is equal to adopted version. 
+    In this case each field of `BlockVersionModifier` from proposal must be either `Nothing` or 
+    `Just` with the same value from `BlockVersionData` of last adopted version, 
+    i.e. we forbid changing any protocol constants. 
+  * Block version from proposal is equal to one from another proposal. 
+    `BlockVersionModifier`s from them must be the same. 
+     If several proposals with the same block version exist, then all `BlockVersionModifier` must be equal.
+  * Block version from proposal is new one. 
+    In this case we must check `BlockVersionModifier` from the proposal is consistent with adopted `BlockVersionData`.  
+    The following checks take place:
+    * script version from proposal must be equal to or greater by `1` adopted one.
+    * max block size from proposal is at most two times greater than adopted one.
+    * unlock stake epoch must not be changed if boostrap era is over
```

Comment:
I don't like your punctuation. Two sentences have a dot in the end, but this one doesn't. For some reason there are no capital letters after dots. It looks strange to me, though I don't know English punctuation well.